pkg/controllers/corecluster/corecluster_controller.go

```diff
@@ -81,38 +87,89 @@ func (r *CoreClusterController) SetupWithManager(mgr ctrl.Manager) error {
 }
 
 // Reconcile reconciles the core cluster object for the openshift-cluster-api namespace.
+//
+//nolint:funlen
 func (r *CoreClusterController) Reconcile(ctx context.Context, req reconcile.Request) (ctrl.Result, error) {
 	logger := log.FromContext(ctx).WithName(controllerName)
 	logger.Info("Reconciling core cluster")
 	defer logger.Info("Finished reconciling core cluster")
 
+	var err error
+
+	co, err := r.GetOrCreateClusterOperator(ctx)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to get or create cluster operator: %w", err)
+	}
+
+	// Populate conditions from the existing cluster operator status.
+	conditions := &co.Status.Conditions
+
+	defer func() {
+		// This function runs before every return, to ensure the controller conditions
+		// for that codepath are correctly synced to the cluster operator status.
+		if syncErr := r.SyncControllerConditions(ctx, co, controllerName, conditions,
+			[]string{CoreClusterControllerAvailableCondition, CoreClusterControllerDegradedCondition}); syncErr != nil {
+			if err != nil {
+				// The sync of the controller conditions failed but there is also a non nil reconcile error.
+				// Log the sync error and pass down the reconciler one.
+				logger.Error(syncErr, "failed to sync cluster operator status")
+			} else {
+				// If there is no reconcile error, but there is a sync error, then we want to propagate the
+				// sync error to be returned by the Reconcile function, so that it will cause an immediate requeue/retry.
+				err = syncErr
```

Comment:
Is this the better way to achieve this vs named return? I thought with named return the final return could be `nil`, but, if this works, and it's commented, I don't have a strong opinion, just curious if both paths work

Reply:
I think this is easier to reason about, as one doesn't need to understand ordering of nil vs named returns.